### PR TITLE
#202 Project Table Filter Component

### PR DESCRIPTION
**Pull request by Ari-S-123**
We implemented and tested all the features for the project table filter component we built for ticket #202. 
Screenshot (706)

### CODE REVIEW

File: src/components/projects/projects-table/projects-table.module.css
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.module.css:8]: you might be able to shorten this with the single line padding that goes something like `padding: 5px 5px 5px 5px`

File: src/components/projects/projects-table/projects-table.test.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.test.tsx:85]: Good catch!

File: src/components/projects/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.tsx:51]: use double equals whenever appropriate. here the linter complains

File: src/components/projects/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.tsx:127]: very soon the `project.projectLead` field will be an array

File: src/components/projects/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.tsx:164]: Pretty sure you should be using the page title component here

File: src/components/projects/projects-table/projects-table/projects-table.module.css
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table/projects-table.module.css:6]: I don't think you should leave empty CSS definitions. Having a CSS file which contains nothing other than the license comment at the top is totally fine

File: src/components/projects/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.tsx:150]: why do it here instead of inline on line 176?

**Review by jamescd18 — CHANGES_REQUESTED** [review]
This is a really great start! I've highlighted the things that stood out to me the most, but this will definitely need a couple more rounds of review and discussion before it's ready to merge. It's really awesome to see you two learning a lot and going beyond your comfort zone to build the component with state management in mind from the ground up. a few things:
- The year created is no longer relevant so that whole field and everything associated should be removed. sorry I didn't catch that and have it removed from the designs sooner
- are all the test IDs absolutely necessary for the testing? remember that the philosophy of testing react components with this testing library is "test what the user can see" and the user can't see test IDs. test IDs are effectively escape hatches to get away from that philosophy.
- remember to consider the naming of variables so that it's intelligible and makes sense
- remember to check the results of the linter (displays inline in the PR code changes tab)
- are there small pieces of abstraction that can be done? like you have a ton of dropdown items for "none", and the dropdown items for car number could perhaps be rewritten as `[1, 2].map(num => <Dropdown.Item ...... >)`?

As always, reply to inline comments, resolve inline comments, start a conversation on slack, and/or ping me when it's ready for another review! Great work so far!!

File: src/components/projects/projects-table/projects-table.tsx
Comment (Ari-S-123) [reply src/components/projects/projects-table/projects-table.tsx:119]: I think we will have to wait until that other branch that deals with that is merged to make the changes in this line.

File: src/components/projects/projects-table/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table/projects-table.tsx:6]: pretty sure that if you have prettier enabled this change shouldn't happen

File: src/components/projects/projects-table/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table/projects-table.tsx:13]: same. shouldn't happen

File: src/custom.scss
Comment (jamescd18) [inline src/custom.scss:7]: why? why on line 7?

File: src/custom.scss
Comment (jamescd18) [inline src/custom.scss:1]: needs license comment

File: src/index.tsx
Comment (jamescd18) [inline src/index.tsx:6]: why make it the first one?

File: src/components/projects/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.tsx:44]: why string name equality when you really want ID equality?

File: src/components/projects/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.tsx:41]: same question, why string equality? it should be ID-based

File: src/components/projects/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.tsx:154]: why not filter first and then transform? that would enable you to have more info for when filtering

File: src/components/projects/projects-table/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table/projects-table.tsx:124]: might be able to remove these now that it's just one element inside

File: src/components/projects/projects-table/projects-table-filter/projects-table-filter.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table-filter/projects-table-filter.tsx:8]: please put this import line first

**Review by jamescd18 — CHANGES_REQUESTED** [review]
This is pretty good! There's a design pattern question I included about filtering first instead of transforming to display projects first. I think filtering first is probably better. Also, I'd love it if you could explain more about why sass is required for just one button. A few more minor items here and there, and I'd love to see new screenshots added to a new comment in the PR.

**Comment by shanejpark** [comment]
> This is pretty good! There's a design pattern question I included about filtering first instead of transforming to display projects first. I think filtering first is probably better. Also, I'd love it if you could explain more about why sass is required for just one button. A few more minor items here and there, and I'd love to see new screenshots added to a new comment in the PR.

The only way to have custom colors for bootstrap objects is to either use !important or sass. The custom color in the sass file is useful if we're going to use more bootstrap objects later on or plan to modify them in any other way. There already are a lot of bootstrap elements in our project and it would be better if all of them just had a "variant: ner-red" line instead of having custom css or using !important.

File: src/custom.scss
Comment (shanejpark) [reply src/custom.scss:7]: You have to modify the variables before importing, or else they'll be loaded without any modification

File: package.json
Comment (jamescd18) [inline package.json:118]: Don't update versions unless justified / required as it could cause mysterious problems

File: package.json
Comment (jamescd18) [inline package.json:131]: so is the verdict that yes you do want to use sass?

File: src/custom.scss
Comment (jamescd18) [inline src/custom.scss:8]: if you're gonna add one, please add the others from [the Brand Guideline doc](https://docs.google.com/document/d/1bV-TGorNeGbn7BhrwP9_nhSb_qmfs5wxJeJY5AfekDo/edit)

**Review by jamescd18 — CHANGES_REQUESTED** [review]
So the verdict is to go ahead and use sass?

File: src/components/projects/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/components/projects/projects-table/projects-table.tsx:145]: I think "seenList" would be more descriptive

**Review by jamescd18 — CHANGES_REQUESTED** [review]

File: src/components/projects/projects-table/projects-table.tsx
Comment (Ari-S-123) [reply src/components/projects/projects-table/projects-table.tsx]: Those changes seem to be old and don't reflect the current state of that file.

**Review by jamescd18 — APPROVED** [review]
This is sooooooooooo awesome to finally see completed! Amazing job guys! I love to see the perseverance and dedication to refining your development skills through trial and error here. Great work!

**Pull request merged**
